src/inputs/input/input.jsx

```diff
@@ -10,100 +10,188 @@ import ClassNames from 'classnames';
 import InputMasked from 'react-text-mask';
 import PropTypes from 'prop-types';
 import React from 'react';
-import ReactDOM from 'react-dom';
+import {
+    UI_CLASS_NAME,
+    BEM_INPUT,
+} from '../../global/constants';
 import Icon from '../../dataDisplay/icon';
 import withStyles from '../../styles/withStyles';
 
 const propTypes = {
+    /**
+     * The actions container that should only contain a Button.
+     */
+    actions: PropTypes.node,
+    /**
+     * Hint for form autofill feature.
+     */
     autoComplete: PropTypes.oneOf(['off', 'on']),
+    /**
+     * Automatically focus the form control when the page is loaded.
+     */
     autoFocus: PropTypes.bool,
+    /**
+     * Assign additional class names to Input.
+     */
+    className: PropTypes.string,
     /**
      * Override or extend the styles applied to Input.
      */
     classes: PropTypes.shape({
+        actions: PropTypes.string,
         root: PropTypes.string,
         hasError: PropTypes.string,
-        hasIcon: PropTypes.string,
+        hasActions: PropTypes.string,
+        hasValue: PropTypes.string,
+        inputContainer: PropTypes.string,
         isDisabled: PropTypes.string,
         isFluid: PropTypes.string,
         isFocused: PropTypes.string,
-        isLoading: PropTypes.string,
         isNumberType: PropTypes.string,
+        isRequired: PropTypes.string,
+        isRequirementComplete: PropTypes.string,
+        isRequirementIncomplete: PropTypes.string,
+        isThemeActiveConstrast: PropTypes.string,
     }).isRequired,
-    className: PropTypes.string,
     /**
-     * An Input can be disabled.
+     * A Input can be disabled.
```

Comment:
Nope: _An_ Input was correct! ;-)
Alternatively, we can go with _the_ Input everywhere to avoid A/An issue.

Reply:
Fixed.

Reply:
Looks good.  Resolving.